querygen/querygen.go

```diff
@@ -57,3 +57,33 @@ func GenBuildPipelineRunStartedQuery(index string) string {
 	)
 	return q
 }
+
+// GenClairScanCompletedQuery returns a Splunk query for generating Segment events
+// when the clair-scan task completes.
+func GenClairScanCompletedQuery(index string) string {
+	q, _ := UserJourneyQueryGen(
+		index,
+		`verb=update `+
+			`"responseStatus.code"=200 `+
+			`"objectRef.apiGroup"="tekton.dev" `+
+			`"objectRef.resource"="taskruns" `+
+			`"objectRef.subresource"="status" `+
+			`"requestObject.metadata.labels.tekton.dev/pipelineTask"="clair-scan" `+
+			`"responseObject.status.completionTime"="*" `+
+			`| spath path=responseObject.status.conditions{} output=conditions `+
+			`| mvexpand conditions `+
+			`| spath input=conditions `+
+			`| where type="Succeeded" AND reason="Succeeded" AND status="True" `+
+			`| spath path=responseObject.status.taskResults{} output=taskResults `+
+			`| mvexpand taskResults `+
+			`| spath input=taskResults `+
+			`| search name="CLAIR_SCAN_RESULT" `+
+			`| spath input=value `+
+			`| eval vulnerabilities_critical=mvindex('vulnerabilities.critical', 0),
+					vulnerabilities_high=mvindex('vulnerabilities.high', 0),
+					vulnerabilities_medium=mvindex('vulnerabilities.medium', 0),
+					vulnerabilities_low=mvindex('vulnerabilities.low', 0) `,
```

Comment:
Here's a version that's:

1. Much more explicit (at the expense of some repetition). There are no more auto-extracted fields using `spath`.
2. Much safer and more consistent with field naming (e.g. `responseObject.status.conditions{}.type` vs `type`).
3. Uses `search` instead of `where`.
4. Specifies the `event` field to bypass the default event naming behavior in the `splunk-to-segment.sh` script.

```suggestion
			`| spath responseObject.status.conditions{} `+
			`| mvexpand responseObject.status.conditions{} `+
			`| search responseObject.status.conditions{}.type="Succeeded" `+
			`responseObject.status.conditions{}.reason="Succeeded" `+
			`responseObject.status.conditions{}.status="True" `+
			`| spath responseObject.status.taskResults{} `+
			`| mvexpand responseObject.status.taskResults{} `+
			`| search responseObject.status.taskResults{}.name="CLAIR_SCAN_RESULT" `+
			`| spath input=responseObject.status.taskResults{}.value path=vulnerabilities.critical output=clair_scan_result.vulnerabilities.critical `+
			`| spath input=responseObject.status.taskResults{}.value path=vulnerabilities.high output=clair_scan_result.vulnerabilities.high `+
			`| spath input=responseObject.status.taskResults{}.value path=vulnerabilities.medium output=clair_scan_result.vulnerabilities.medium `+
			`| spath input=responseObject.status.taskResults{}.value path=vulnerabilities.low output=clair_scan_result.vulnerabilities.low `+
			`| eval event="Clair scan TaskRun completed"`,
```

This necessitates a change to the `srcFields` in the fieldset as well.

Reply:
@amisstea I have now updated the query with your suggestions and compared the result. It works.